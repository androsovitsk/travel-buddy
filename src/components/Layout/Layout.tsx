import React from 'react'
import Grid from '@mui/material/Unstable_Grid2'
import Paper from '@mui/material/Paper'
// @ts-ignore
import background from '../../assets/background.jpg'

const Layout: React.FC<React.PropsWithChildren> = ({ children }) => {
  return (
    <Grid
      container
      sx={{
        height: '100vh',
        backgroundImage: `url(${background})`,
        backgroundSize: 'cover',
        backgroundPosition: 'center'
      }}
      justifyContent={'center'}
      alignItems={'center'}
    >
      <Paper
        elevation={3}
        sx={{
          width: '726px',
          padding: (theme) => `${theme.spacing(6)} ${theme.spacing(12)}`
        }}
      >
        {children}
      </Paper>
    </Grid>
  )
}

Layout.displayName = 'Layout'

export default Layout
