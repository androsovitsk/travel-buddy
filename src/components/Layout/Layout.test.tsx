import { render, screen } from '@testing-library/react'
import Layout from './Layout'

describe('Layout', () => {
  it('should match the snapshot', () => {
    const view = render(<Layout />)

    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should render the children', () => {
    const testId = 'test-id'

    render(
      <Layout>
        <div data-testid={testId} />
      </Layout>
    )

    const childrenElement = screen.getByTestId(testId)
    expect(childrenElement).toBeInTheDocument()
  })
})
