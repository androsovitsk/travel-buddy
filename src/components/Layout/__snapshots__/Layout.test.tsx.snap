// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-pc6395-MuiGrid2-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1odm8bn-MuiPaper-root"
    />
  </div>
</DocumentFragment>
`;
